api/filters/replacement/replacement.go

```diff
@@ -39,6 +39,13 @@ func (f Filter) Filter(nodes []*yaml.RNode) ([]*yaml.RNode, error) {
 }
 
 func getReplacement(nodes []*yaml.RNode, r *types.Replacement) (*yaml.RNode, error) {
+	if r.SourceValue != nil && r.Source != nil {
```

Comment:
I think we already have check while reading yaml nodes to not allow nil while adding elements to `Replacements` list. But can we do an early return if `r == nil` to avoid nil pointers just in case?

Reply:
Actually this can be done in `Filter` itself.

Reply:
Sorry but I've doubts on that, when you say an early return, it should return an error?
So in case one of the replacements given is nil, it should return an error saying that a null replacement is not valid, or it should ignore it and continue with the loop?

Reply:
I don't know if I'm understanding it. 
In `Filter` function, replacements is an slice of type `[]types.Replacement`, so their elements should not be nil.
Inside `getReplacement` function, as I can see, it's passed by reference so it can modify the field path to the default one in case it is not given. 
As I see it, inside the for loop on `Filter` function, `r` cannot be nil, don't know if I'm missing something, sorry 😅